stdlib/public/core/FloatingPointTypes.swift.gyb

```diff
@@ -90,6 +90,12 @@ public struct ${Self} {
 ${Availability(bits)}
 extension ${Self}: CustomStringConvertible {
   /// A textual representation of the value.
+  ///
+  /// For all finite values, the value of this property is a string that can be
```

Comment:
This is a guarantee for all types that conform to `LosslessStringConvertible`—I think it would be nice to harmonize the text with what’s there, and perhaps even reference the protocol explicitly.

Reply:
The documentation for `LosslessStringConvertible` is far from ideal.  It currently states "... it should be possible to re-create an instance from its string representation."  That sentence has two problems:  First, it's not binding (it's "should", not "must").  More importantly, it's not clear what it means to "re-create an instance."  (Does that mean that the new instance is `==` to the old one?  That would require that `LosslessStringConvertible` conform to `Equatable`.)

Deferring to that here is problematic because of NaN values.  NaN values can't be compared for equality (so it's not easy to define what it means to "re-create an instance"), and the current `description` property doesn't distinguish different NaN values.